dex/farm/src/lib.rs

```diff
@@ -155,16 +156,27 @@ pub trait Farm:
         Wrapper::<Self>::calculate_rewards(
             self,
             user,
+            farm_token_nonce,
             &farm_token_amount,
             &attributes,
             &storage_cache,
         )
     }
 
+    #[payable("*")]
+    #[endpoint(mergeFarmTokensWithEnergy)]
+    fn merge_farm_tokens_with_energy_endpoint(&self) -> EsdtTokenPayment<Self::Api> {
+        let caller = self.blockchain().get_caller();
+        let new_tokens = self.merge_farm_tokens_with_energy(&caller);
+        self.send_payment_non_zero(&caller, &new_tokens);
+        new_tokens
+    }
+
     #[payable("*")]
     #[endpoint(mergeFarmTokens)]
     fn merge_farm_tokens_endpoint(&self) -> EsdtTokenPayment<Self::Api> {
```

Comment:
There should not be two merge endpoints at all. The old endpoint is enough, with the OptionalArg of original caller. You would need to update energy in both cases either way.

Reply:
Done.